solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java

```diff
@@ -90,18 +91,24 @@ public class LBHttp2SolrClient extends LBSolrClient {
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
   private final Http2SolrClient solrClient;
 
+  private final LongConsumer delayedRequestListener;
+
   /**
    * @deprecated Use {@link LBHttp2SolrClient.Builder} instead
    */
   @Deprecated
   public LBHttp2SolrClient(Http2SolrClient solrClient, String... baseSolrUrls) {
     super(Arrays.asList(baseSolrUrls));
     this.solrClient = solrClient;
+    this.delayedRequestListener = DELAYED_REQ_LOGGER;
   }
 
-  private LBHttp2SolrClient(Http2SolrClient solrClient, List<String> baseSolrUrls) {
+  private LBHttp2SolrClient(
+      Http2SolrClient solrClient, List<String> baseSolrUrls, LongConsumer delayedRequestListener) {
     super(baseSolrUrls);
     this.solrClient = solrClient;
+    this.delayedRequestListener =
+        delayedRequestListener == null ? DELAYED_REQ_LOGGER : delayedRequestListener;
```

Comment:
Looks like now it will not log the request? Logging might give more context to debug?

Reply:
we could do that too